lib/fastly/client.rb

```diff
@@ -102,6 +102,11 @@ def purge(url, params = {})
       extras = params.delete(:headers) || {}
       uri    = URI.parse(url)
       http   = Net::HTTP.new(uri.host, uri.port)
+
+      if uri.is_a? URI::HTTPS
```

Comment:
sure happy to make that change.  for what its worth I copied this technique from line 23 of the same file...